### PR TITLE
cmd/password/list: show TTL remaining and if password is expired

### PR DESCRIPTION
**Pull request by mattrobenolt**
This adds two new columns to the password list, "TTL REMAINING", and "EXPIRED".

This allows a user to see, if they have a credential that is ephemeral, to see how long it has left before it's expired, and if it is expired already.

Relies on https://github.com/planetscale/planetscale-go/pull/193

After creating with a 60s TTL:

```
$ ./pscale password list mattdb
  ID             NAME                     BRANCH   USERNAME               ROLE     ROLE DESCRIPTION               TTL   TTL REMAINING   EXPIRED
 -------------- ------------------------ -------- ---------------------- -------- ------------------------------ ----- --------------- ---------
  zxc6td6uu0rh   test-ttl                 main     amzdunhtul1jou6yd11r   reader   Can Read                        60              57   No
  oddkvx4zj8y9   main-2023-11-11-e6qg9j   main     cc7elecd13x7gokz30sh   admin    Can Read, Write & Administer     0               0   No
```

we see both an ephemeral password with a 60s TTL and 57s remaining, but not expired, and a password with a 0 TTL, meaning it has no expiration.

After waiting 60s, we see:

```
$ ./pscale password list mattdb
  ID             NAME                     BRANCH   USERNAME               ROLE     ROLE DESCRIPTION               TTL   TTL REMAINING   EXPIRED
 -------------- ------------------------ -------- ---------------------- -------- ------------------------------ ----- --------------- ---------
  zxc6td6uu0rh   test-ttl                 main     amzdunhtul1jou6yd11r   reader   Can Read                        60               0   Yes
  oddkvx4zj8y9   main-2023-11-11-e6qg9j   main     cc7elecd13x7gokz30sh   admin    Can Read, Write & Administer     0               0   No
```

### CODE REVIEW

**Comment by mattrobenolt** [comment]
I'm open to renaming these columns since TTL REMAINING adds quite a bit to the UI when there's this is very often going to be single digit.

Would be nice to conditionally show columns based on if any have any TTLs or not and not even show TTL, TTL REMAINING, and EXPIRED, if nothing can expire.

**Comment by dbussink** [comment]
I like the idea of not showing the column at all if nothing expires.

**Comment by mattrobenolt** [comment]
I've updated it so only the last 3 columns `(TTL, TTL REMAINING, EXPIRED)` only show up if any password is ephemeral. If they are all long lived, we hide these.

File: internal/mock/password.go [hunk 1]
```diff
@@ -36,3 +38,8 @@ func (b *PasswordsService) Delete(ctx context.Context, req *ps.DeleteDatabaseBra
 	b.DeleteFnInvoked = true
 	return b.DeleteFn(ctx, req)
 }
+
+func (b *PasswordsService) Renew(ctx context.Context, req *ps.RenewDatabaseBranchPasswordRequest) (*ps.DatabaseBranchPassword, error) {
+	b.DeleteFnInvoked = true
```
[the remark this answers is not in the document]
Comment (mattrobenolt) [reply internal/mock/password.go:43]: Oh, oops.

**Comment by mdlayher** [comment]
A thought: It'd be dope if we could render the TTL as a duration in the output (may be more annoying given the structs), I also put in support for creating passwords using a duration with https://github.com/planetscale/cli/pull/819.

**Comment by mattrobenolt** [comment]
I'd be down to follow up on that, I would like to get this shipped as-is for now since there's a lot building off of this.

**Pull request merged**